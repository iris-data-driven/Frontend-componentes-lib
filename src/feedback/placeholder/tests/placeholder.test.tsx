import React from 'react';
import { render, screen } from '@testing-library/react';

import Placeholder from '../';

describe('Placeholder', () => {
  it('Should render correctly without an implicit type', () => {
    render(<Placeholder />);
    const component = screen.getByRole('placeholder-loader');
    expect(component).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

  it('Should render correctly with heading type', () => {
    render(<Placeholder type="heading" />);
    const component = screen.getByRole('placeholder-loader');
    expect(component).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

  it('Should render correctly with circle type', () => {
    render(<Placeholder type="circle" />);
    const component = screen.getByRole('placeholder-loader');
    expect(component).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

  it('Should render correctly with custom type', () => {
    render(<Placeholder type="custom" />);
    const component = screen.getByRole('placeholder-loader');
    expect(component).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });
});
