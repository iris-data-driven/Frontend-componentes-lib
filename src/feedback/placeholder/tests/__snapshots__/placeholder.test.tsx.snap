// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Should render correctly with circle type 1`] = `
<div
  aria-busy="true"
  class="sc-AxjAm buxUmO"
  role="placeholder-loader"
  type="circle"
/>
`;

exports[`Placeholder Should render correctly with custom type 1`] = `
<div
  aria-busy="true"
  class="sc-AxjAm cjwarf"
  role="placeholder-loader"
  type="custom"
/>
`;

exports[`Placeholder Should render correctly with heading type 1`] = `
<div
  aria-busy="true"
  class="sc-AxjAm hiZUKR"
  role="placeholder-loader"
  type="heading"
/>
`;

exports[`Placeholder Should render correctly without an implicit type 1`] = `
<div
  aria-busy="true"
  class="sc-AxjAm kNpHOQ"
  role="placeholder-loader"
  type="text"
/>
`;
